server/src/main/java/com/android/identity/wallet/server/VerifierServlet.kt

```diff
@@ -1436,7 +1469,7 @@ private fun calcClientMetadata(session: Session, format: String): JSONObject {
     val client_metadata = JSONObject()
     client_metadata.put("authorization_encrypted_response_alg", "ECDH-ES")
     client_metadata.put("authorization_encrypted_response_enc", "A128CBC-HS256")
-    client_metadata.put("response_mode", "direct_post.jwt")
+    client_metadata.put("response_mode", session.responseMode ?: "direct_post.jwt")
```

Comment:
Can't see the whole picture, but `?:` operator is a frequent source of errors when confused with the java `?:` 
operator. I saw that now we have 2 distinct values for that mode, so just the null check might be inappropriate (just something I would personally doublecheck understanding the code and intent here better).